pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -730,22 +730,29 @@ public String getActualTableName(String tableName) {
    */
 
   /**
-   * Returns the segments for the given table.
+   * Returns the segments for the given table from the ideal state.
    *
    * @param tableNameWithType Table name with type suffix
    * @param shouldExcludeReplacedSegments whether to return the list of segments that doesn't contain replaced segments.
    * @return List of segment names
    */
   public List<String> getSegmentsFor(String tableNameWithType, boolean shouldExcludeReplacedSegments) {
```

Comment:
`getSegmentsFor` is originally also used in `SegmentRelocator` and `RetentionManager` are those 2 usage both IS as well?

Reply:
Yes. Checked all the usage of them and moved the ones to clean up the orphan segments (e.g. when deleting the table or all segments) to use `getSegmentsFromPropertyStore()`